File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/RefreshQ.java
package Auxiliary_Modules;

public class RefreshQ extends UniqueItemSizeLimitedPriorityQueue{
    public void removeItem(String item){
        super.elements.remove(item);
    }

    @Override
    public void add(String item) {
        // FILO
        if (super.contains(item)){
            removeItem(item);
        }
        super.add(item);
    }
    public void stuff(String data) {
        // FILO 1st in last out
        if (elements.size() == getLimit()) {
            poll();
        }
        elements.add(data);
    }

}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/RefreshQ.kt
package auxiliary_modules

class RefreshQ : UniqueItemSizeLimitedPriorityQueue() {
    override fun removeItem(item: String) {
        super.elements.remove(item)
    }

    override fun add(item: String) {
        // FILO
        if (super.contains(item)) {
            removeItem(item)
        }
        super.add(item)
    }
    fun stuff(data: String?) {
        // FILO 1st in last out
        if (elements.size === limit) {
            poll()
        }
        elements.add(data!!)
    }
}